play-json/jvm/src/test/scala/play/api/libs/json/JsonSpec.scala

```
@@ -69,6 +70,27 @@ class JsonSpec extends org.specs2.mutable.Specification {
 
   val mapper = new ObjectMapper()
 
+  val preserveZeroDecimal: JacksonJson = {
+    val defaultSerializerSettings = JsonParserSettings.settings.bigDecimalSerializerSettings
+    val defaultParserSettings     = JsonParserSettings.settings.bigDecimalParseSettings
+    val serializerSettings = BigDecimalSerializerSettings(
+      defaultSerializerSettings.minPlain,
+      defaultSerializerSettings.maxPlain,
+      preserveZeroDecimal = true
+    )
+    JacksonJson(JsonParserSettings(defaultParserSettings, serializerSettings))
+  }
+
+  def withJacksonJson[T](jacksonJson: JacksonJson)(f: () => T) = {
+    val oldInstance = JacksonJson.set(jacksonJson)
+    try {
+      JacksonJson.get = jacksonJson
+      f.apply()
+    } finally {
```

Silent error is generally not a good idea, even in tests